gtsfm/common/gtsfm_data.py

```diff
@@ -0,0 +1,196 @@
+"""Class to hold the tracks and cameras of a 3D scene.
+
+Authors: Ayush Baid
+"""
+import itertools
+from typing import Dict, List, Optional
+
+import numpy as np
+from gtsam import PinholeCameraCal3Bundler, SfmTrack
+
+import gtsfm.utils.graph as graph_utils
+
+EQUALITY_TOLERANCE = 1e-5
+
+
+class GtsfmData:
+    """Class containing cameras and tracks, essentially describing the complete 3D scene.
+    
+    This class is needed over GTSAM's SfmData type because GTSAM's type does not allow for non-contiguous cameras.
+    The situation of non-contiguous cameras can exists because of failures in front-end.
+    """
+
+    def __init__(self, number_images: int) -> None:
```

Comment:
why does this class need to know `number_images`?

Reply:
Because we can have 5 images, but just 3 cameras optimized by SfM

Reply:
right. is it required information inside the class though? i.e. could we do without it?